### ISSUE TITLE
Fails when test method default values are given as consts

### ISSUE THREAD

**Issue by SignpostMarv** [psalm/phpunit-psalm-plugin#25]
Just wondering what the best approach is to using default arguments on a test method & documenting the return type with a data provider, as I'm currently getting `TooFewArguments .... expecting at least 3, but saw 2 provided by .... ::dataprovider_markuparraytomarkupstring():(array<int, array{0:string, 1:array<array-key, mixed>}>)`, where the test method is something like:

```php
public function test_thing(
    string $expected,
    array $source,
    bool $as_xml = SomeOtherClassBecauseMutationTesting::DEFAULT_BOOL_AS_XML,
    int $flags = Etc::DEFAULT_YOU_GET_THE_IDEA) {
...
}
```

and the data provider has an array return but only a couple of the inner arrays include some of the extra args?

I'm thinking typed args is better than shoving in a mixed variadic & when I tried doing `@psalm-return array<int, array{0:string, 1:array, 2?:bool, 3?:int}>` it didn't exactly work as expected.

**Comment by weirdan** [comment]
> when I tried doing `@psalm-return array<int, array{0:string, 1:array, 2?:bool, 3?:int}>` it didn't exactly work as expected

It is supposed to work just like that, see these tests: https://github.com/psalm/phpunit-psalm-plugin/blob/e33ae73dc725acdf361645625667c65aa88731ee/tests/acceptance/TestCase.feature#L585-L652

If it doesn't, it's a bug.

With that said, your original approach with defaults is also supposed to work (and does, when defaults are specified as literal values). However there appears to be a problem when constants are used as default values.

So here's the test I'm going to add and fix this issue:
```cucumber
  Scenario: Provider not returning some offsets is fine when test method has defaults for those params (specified as constants)
    Given I have the following code
      """
      class MyTestCase extends TestCase
      {
        /** @var string */
        const S = "s";

        /** @return iterable<string,array{0:int,1?:string}> */
        public function provide() {
          yield "data set name" => rand(0,1) ? [1] : [1, "ss"];
        }

        /**
         * @return void
         * @dataProvider provide
         */
        public function testSomething(int $int, string $_str = self::S) {
          $this->assertEquals(1, $int);
        }
      }
      """
    When I run Psalm
    Then I see no errors
```

**Comment by SignpostMarv** [comment]
I'll hopefully give it a check tomorrow (is dayjob stuff)

**Comment by weirdan** [comment]
I have a fix for const issue already working locally. Barring any possible compatibility issues, you'll be able to test it by then.

**Comment by weirdan** [comment]
Fixed in #26
